services/api/src/resources/environment/harborSetup.js

```diff
@@ -0,0 +1,136 @@
+// @flow
+
+/* ::
+import type MariaSQL from 'mariasql';
+*/
+
+const harborClient = require('../../clients/harborClient');
+const logger = require('../../logger');
+const R = require('ramda');
+const Sql = require('../env-variables/sql');
+const { isPatchEmpty, prepare, query, whereAnd } = require('../../util/db');
+
+const lagoonHarborRoute = R.compose(
+  R.defaultTo('http://172.17.0.1:8084'),
+  R.find(R.test(/harbor-nginx/)),
+  R.split(','),
+  R.propOr('', 'LAGOON_ROUTES'),
+)(process.env);
+
+const createHarborOperations = (sqlClient /* : MariaSQL */) => ({
+  addEnvironment: async (openshiftProjectName, environmentID) => {
+    // Create harbor project
+    try {
+      var res = await harborClient.post(`projects`, {
+        body: {
+          count_limit: -1,
+          project_name: openshiftProjectName,
+          storage_limit: -1,
+          metadata: {
+            auto_scan: "true",
+            reuse_sys_cve_whitelist: "true",
+            public: "false"
+          }
+        }
+      });
+      logger.debug(`Harbor project ${openshiftProjectName} created!`)
+    } catch (err) {
+      // 409 means project already exists
+      // 201 means project created successfully
+      if (err.statusCode == 409) {
+        logger.info(`Unable to create the harbor project "${openshiftProjectName}", as it already exists in harbor!`)
+      } else {
+        console.log(res)
+        logger.error(`Unable to create the harbor project "${openshiftProjectName}" !!`, err)
+      }
+    }
+
+    // Get new harbor project's id
```

Comment:
it gets not only new harbor project ids? but also existing ones?

Reply:
That is correct; we want to know if any harbor project exists for this lagoon project name